lib/pc_schema.c

```diff
@@ -635,3 +635,36 @@ pc_schema_get_size(const PCSCHEMA *s)
 {
 	return s->size;
 }
+
+/**
+* Return true if s1 and s2 have the same dimensions at the same
+* positions, otherwise return false.
+*/
+uint32_t
+pc_schema_similar(const PCSCHEMA *s1, const PCSCHEMA *s2)
+{
+	size_t i;
+
+	if ( s1->ndims != s2->ndims )
+		return PC_FALSE;
+
+	for ( i = 0; i < s1->ndims; i++ )
+	{
+		PCDIMENSION *s1dim = s1->dims[i];
+		PCDIMENSION *s2dim = s2->dims[i];
+
+		if ( strcmp(s1dim->name, s2dim->name) != 0 )
+			return PC_FALSE;
+
+		if ( s1dim->interpretation != s2dim->interpretation )
+			return PC_FALSE;
+
+		if ( s1dim->scale != s2dim->scale )
+			return PC_FALSE;
+
+		if ( s1dim->offset != s2dim->offset )
+			return PC_FALSE;
```

Comment:
If we do that then the data won't be reinterpreted if just the offsets/scales change in the new schema. I do have use-cases where this would a problem.

Currently, the fast path is always used when the schemas are deemed equivalent, whathever the value of `reinterpret`. We could change the logic to skip the fast path and force the re-interpretation when `reinterpret` is `TRUE`, but I am not sure. Is that what you're suggesting?